homeassistant/components/neato/config_flow.py

```diff
@@ -1,112 +1,34 @@
-"""Config flow to configure Neato integration."""
-
+"""Config flow for Neato Botvac."""
 import logging
 
-from pybotvac import Account, Neato, Vorwerk
-from pybotvac.exceptions import NeatoLoginException, NeatoRobotException
-import voluptuous as vol
-
 from homeassistant import config_entries
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import config_entry_oauth2_flow
 
 # pylint: disable=unused-import
-from .const import CONF_VENDOR, NEATO_DOMAIN, VALID_VENDORS
+from .const import NEATO_DOMAIN
 
 DOCS_URL = "https://www.home-assistant.io/integrations/neato"
-DEFAULT_VENDOR = "neato"
 
 _LOGGER = logging.getLogger(__name__)
 
 
-class NeatoConfigFlow(config_entries.ConfigFlow, domain=NEATO_DOMAIN):
-    """Neato integration config flow."""
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=NEATO_DOMAIN
+):
+    """Config flow to handle Neato Botvac OAuth2 authentication."""
 
-    VERSION = 1
+    VERSION = 2
+    DOMAIN = NEATO_DOMAIN
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
-    def __init__(self):
-        """Initialize flow."""
-        self._username = vol.UNDEFINED
-        self._password = vol.UNDEFINED
-        self._vendor = vol.UNDEFINED
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
 
-    async def async_step_user(self, user_input=None):
-        """Handle a flow initialized by the user."""
-        errors = {}
+    async def async_oauth_create_entry(self, data: dict) -> dict:
```

Comment:
We can define the user step instead, check for current entries, then call the super user step defined in the parent.

Reply:
Done, thank you

Reply:
What's the reason behind this? Check for current entries to abort with `already_configured`? Why can't we do that in `async_oauth_create_entry`.

Reply:
It's less code duplication. The overriden `async_oauth_create_entry` already does what we do here, besides checking for existing entries.

Reply:
Thanks, done